pkg/userdata/rhel/provider.go

```diff
@@ -0,0 +1,320 @@
+/*
+Copyright 2019 The Machine Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//
+// UserData plugin for RHEL.
+//
+
+package rhel
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"text/template"
+
+	"github.com/Masterminds/semver"
+
+	"github.com/kubermatic/machine-controller/pkg/apis/plugin"
+	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
+	userdatahelper "github.com/kubermatic/machine-controller/pkg/userdata/helper"
+)
+
+// Provider is a pkg/userdata/plugin.Provider implementation.
+type Provider struct{}
+
+// UserData renders user-data template to string.
+func (p Provider) UserData(req plugin.UserDataRequest) (string, error) {
+	tmpl, err := template.New("user-data").Funcs(userdatahelper.TxtFuncMap()).Parse(userDataTemplate)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse user-data template: %v", err)
+	}
+
+	kubeletVersion, err := semver.NewVersion(req.MachineSpec.Versions.Kubelet)
+	if err != nil {
+		return "", fmt.Errorf("invalid kubelet version: '%v'", err)
+	}
+
+	pconfig, err := providerconfigtypes.GetConfig(req.MachineSpec.ProviderSpec)
+	if err != nil {
+		return "", fmt.Errorf("failed to get provider config: %v", err)
+	}
+
+	if pconfig.OverwriteCloudConfig != nil {
+		req.CloudConfig = *pconfig.OverwriteCloudConfig
+	}
+
+	if pconfig.Network != nil {
+		return "", errors.New("static IP config is not supported with RHEL")
+	}
+
+	rhelConfig, err := LoadConfig(pconfig.OperatingSystemSpec)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse OperatingSystemSpec: '%v'", err)
```

Comment:
Why do we quote the error? (`'%v'`)

Reply:
This is actually a copy and past mistake from this code here: 
https://github.com/kubermatic/machine-controller/blob/master/pkg/userdata/centos/provider.go#L48

Will fix it in both centos and rhel